lib/r10k/puppetfile.rb

```diff
@@ -69,6 +70,16 @@ def set_forge(forge)
     @forge = forge
   end
 
+  # @param [Array<String>] modules
+  def validate_no_duplicate_names(modules)
+    dupes = modules
+            .group_by { |mod| mod.name }
+            .select { |_, v| v.size > 1 }.map(&:first)
+    unless dupes.empty?
+      logger.warn _("Puppetfiles should contain duplicate module names and will result in an error in r10k v3.x. Please remove the remove the duplicates of the following modules:\n%{dupes}" % {dupes: dupes.join("\n")})
```

Comment:
I now think these \n's should be spaces, and this should all be on one line in the log.